src/index.ts

```diff
@@ -165,42 +193,19 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
   }
 
   if (response) {
-    let headers = new Headers(response.headers)
-
-    let shouldRevalidate = false
-    // Four preconditions must be met for a 304 Not Modified:
-    // - the request cannot be a range request
-    // - client sends if-none-match
-    // - resource has etag
-    // - test if-none-match against the pathKey so that we test against KV, rather than against
-    // CF cache, which may modify the etag with a weak validator (e.g. W/"...")
-    shouldRevalidate = [
-      request.headers.has('range') !== true,
-      request.headers.has('if-none-match'),
-      response.headers.has('etag'),
-      request.headers.get('if-none-match') === `${pathKey}`,
-    ].every(Boolean)
-
-
-    if (shouldRevalidate) {
-      // fixes issue #118
+    if (response.status > 300) {
       if (response.body && 'cancel' in Object.getPrototypeOf(response.body)) {
-        response.body.cancel();
+        response.body.cancel()
         console.log('Body exists and environment supports readable streams. Body cancelled')
       } else {
         console.log('Environment doesnt support readable streams')
       }
-
-      headers.set('cf-cache-status', 'REVALIDATED')
-      response = new Response(null, {
-        status: 304,
-        headers,
-        statusText: 'Not Modified',
-      })
+      response = new Response(null, response)
     } else {
-      headers.set('CF-Cache-Status', 'HIT')
-      response = new Response(response.body, { headers })
+      response = new Response(response.body, response)
+      response.headers.set('cf-cache-status', 'HIT')
```

Comment:
It seems that Cloudflare Web Server or Cloudflare Cache will add `cf-cache-status` header itself.

I have removed the line in my own worker script, but the header still appears in the final response.

Reply:
does that header still appear if you remove that resource from cloudflare's cache?

Reply:
To be honest, I’m not sure if the cf-cache-status headers need to be set or not. If they are redundant, I still think they should be retained b/c all the tests will need to be refactored otherwise. Best to discuss this issue on a separate thread IMO

Reply:
> does that header still appear if you remove that resource from cloudflare's cache?

Yes.